Realm/RLMArray.mm

```diff
@@ -299,6 +299,13 @@ - (RLMResults *)objectsWhere:(NSString *)predicateFormat args:(va_list)args
     return [self objectsWithPredicate:[NSPredicate predicateWithFormat:predicateFormat arguments:args]];
 }
 
+- (id)valueForKeyPath:(NSString *)keyPath {
+    if (_backingArray) {
+        return [_backingArray valueForKeyPath:keyPath];
```

Comment:
That's true. I don't think that highlighting that in the documentation will be sufficient. That would be easily missed in development.

Even if it seems to me that `RLMCollectionValueForKey` could easily support key paths as well instead just keys, which would cover the array operators. The numeric operators seem on the first look like they would be most efficiently implemented in Core. But KVC also allows nesting those operators (`@sum.employees.@unionOfObjects.pets.age`), which might not be feasible for such a solution, so that we would end-up using accessors for each numeric key path operator with a nested argument and aggregating the retrieved `NSNumber` values instead. This would mean that we would need to highlight performance considerations.

I'd tend to start with not supporting nested key paths at all consistenly, by checking here as well, whether the key path is nested.